dot/network/service.go

```diff
@@ -487,22 +487,19 @@ func (s *Service) Stop() error {
 		logger.Errorf("Failed to close host: %s", err)
 	}
 
-	s.host.cm.peerSetHandler.Stop()
-
 	// check if closeCh is closed, if not, close it.
-mainloop:
 	for {
 		select {
 		case _, hasMore := <-s.closeCh:
 			if !hasMore {
-				break mainloop
+				return nil
 			}
+		case <-s.closeCh:
+			return nil
```

Comment:
Wait a minuteee.... this looks incredibly wrong i think?

Why do we have two cases pulling from the same channel?